pkg/middleware/middleware.go

```diff
@@ -350,7 +351,10 @@ func (s *SouinBaseHandler) Store(
 		}
 		res.Header.Set(rfc.StoredLengthHeader, res.Header.Get("Content-Length"))
 		response, err := httputil.DumpResponse(&res, true)
-		if err == nil && (bLen > 0 || canStatusCodeEmptyContent(statusCode) || s.hasAllowedAdditionalStatusCodesToCache(statusCode)) {
+		isHeadRequest := rq.Method == http.MethodHead
+		isHeadAllowed := slices.Contains(s.Configuration.GetDefaultCache().GetAllowedHTTPVerbs(), http.MethodHead)
+		canCacheEmptyBody := bLen == 0 && isHeadRequest && isHeadAllowed
+		if err == nil && (bLen > 0 || canCacheEmptyBody || canStatusCodeEmptyContent(statusCode) || s.hasAllowedAdditionalStatusCodesToCache(statusCode)) {
```

Comment:
```suggestion
				if err == nil && (bLen > 0 || rq.Method == http.MethodHead || canStatusCodeEmptyContent(statusCode) || s.hasAllowedAdditionalStatusCodesToCache(statusCode)) {
```

`slices.Contains(s.Configuration.GetDefaultCache().GetAllowedHTTPVerbs(), http.MethodHead)` is already checked at L.674
```
	if !req.Context().Value(context.SupportedMethod).(bool) {
```
and the body length may be greater than or equal to 0 so we don't have to check this.